bots/pr/src/main/java/org/openjdk/skara/bots/pr/ApprovalCommand.java

```diff
@@ -0,0 +1,144 @@
+            return;
+        }
+
+        var arg = command.args().trim().toLowerCase();
+        if (!arg.isEmpty() && !(arg.equals("yes") || arg.equals("y") || arg.equals("no") || arg.equals("n"))) {
+            showHelp(reply);
+            return;
+        }
+
+        if (arg.equals("no") || arg.equals("n")) {
+            for (var vcsIssue : workItem.issues(false, false)) {
+                var issueOpt = bot.issueProject().issue(vcsIssue.shortId());
+                issueOpt.ifPresent(issue -> {
+                    if (!issue.labelNames().contains(workItem.requestLabelName())) {
+                        // The maintainers may disapprove the PR before it is ready.
+                        // The bot should add the fix request label firstly to avoid the strange
+                        // middle state which has disapproval label but has no fix request label.
+                        issue.addLabel(workItem.requestLabelName());
+                    }
+                    if (issue.labelNames().contains(workItem.approvalLabelName())) {
+                        // If the maintainers have approved the PR before,
+                        // the bot should remove the approval label first.
+                        issue.removeLabel(workItem.approvalLabelName());
+                    }
+                    if (!issue.labelNames().contains(workItem.disapprovalLabelName())) {
+                        issue.addLabel(workItem.disapprovalLabelName());
+                    }
+                });
+            }
+            if (pr.labelNames().contains("approval")) {
+                pr.removeLabel("approval");
+            }
+            disapprovalReply(pr, reply);
+            pr.setState(Issue.State.CLOSED);
+            return;
+        }
+
+        pr.setState(Issue.State.OPEN);
+        for (var vcsIssue : workItem.issues(false, false)) {
+            var issueOpt = bot.issueProject().issue(vcsIssue.shortId());
+            issueOpt.ifPresent(issue -> {
+                if (!issue.labelNames().contains(workItem.requestLabelName())) {
+                    // The maintainers may approve the PR before it is ready.
+                    // The bot should add the fix request label firstly to avoid the strange
+                    // middle state which has approval label but has no fix request label.
+                    issue.addLabel(workItem.requestLabelName());
+                }
+                if (issue.labelNames().contains(workItem.disapprovalLabelName())) {
+                    // If the maintainers have disapproved the PR before,
+                    // the bot should remove the disapproval label first.
+                    issue.removeLabel(workItem.disapprovalLabelName());
+                }
+                if (!issue.labelNames().contains(workItem.approvalLabelName())) {
+                    issue.addLabel(workItem.approvalLabelName());
+                }
+            });
+        }
+        if (pr.labelNames().contains("approval")) {
```

Comment:
Also, the label should probably be `approved`, since we use adjectives (like `ready`) for labels.

Reply:
I read the [labels](https://github.com/openjdk/jdk/labels) of the JDK project. Almost all of them are nouns and only few labels are adjectives. And the `approved` may be misunderstood as `the pull request has been approved`. So I use the label `approval`.

Reply:
I agree, the `approval` label is meant to signal that this PR is waiting on/needs approval.

Reply:
Ah, I see. Maybe `needs-approval` then?

Reply:
Or maybe I need to understand the entire labeling workflow better... 

Is this label supposed to signal "this PR is of a kind that needs approval" and all PRs of that kind (i.e., it is a backport made to an update repo) will have that label during their entire life span? Or is it supposed to act like a release blocker (as `ready` but inverted), so when approval is given, then the label is dropped?

I have a hard time figuring out how this PR label relates to the "important" labels in JBS. I think I naively assumed that the PR label should reflect the state of the JBS labels, so you could understand from looking at the PR (and not having to go to JBS) to understand if it was in a state where it needed an approval request in JBS, a JBS approval request had been asked, or an approval had been given or denied in JBS. 

Since that was my mental model, having just a single `approval` label made no sense at all to me.

Reply:
My Interpretation of this label is that it gets added when the PR is entering the request for approval stage and gets removed once approval has been given. That doesn't exactly correspond to any of the JBS labels. The approval process starts when a PR is otherwise ready for integration. This label would then automatically show up to remind the committer that approval is needed. It is then the responsibility of the committer to apply for approval in JBS (either directly or through the new `/request-approval` command). The maintainer then approves (in JBS or with the `/approve` command), and once that is done, the PR is truly ready for integration and the `approval` label is removed.

Alternative definitions would be:

1. Add it to all PRs that need approval for the whole lifecycle. Not really helpful as this is a per repo/branch configuration option. This would also kind of encourage people to try to start the approval process ahead of time.
2. Mimic the JBS label jdkXu-fix-request. That would make it more useful for maintainers looking for PRs to approve, but would not help remind committers that they need to work on the process. Also I don't think Skara should just copy labels from JBS to PRs. 

Similar labels today are `csr` and `jep`. They are added once it's known to Skara that they are needed, either by explicit command or automatic discovery. The `csr` label is removed when that requirement is fulfilled.

Reply:
Ok, fine, if the label is `csr` and not `csr-needed`, then it makes sense to have this as `approval`, not `approval-needed`.

Reply:
How does `approval` interplay with `ready`? Can a PR be both `ready` and `approval` at the same time? Or does it go from `rfr` to `approval` to `ready`? Or something else entirely?

Reply:
The `ready` label should mean that it's now possible to `/integrate`, so that means `rfr` -> `approval` -> `ready` in this case.